File: apps/marketplace-service/internal/model/actor.go
package model

import "time"

type ActorStatus string

const (
	ActorStatusDraft     ActorStatus = "DRAFT"
	ActorStatusPublished ActorStatus = "PUBLISHED"
	ActorStatusDeprecated ActorStatus = "DEPRECATED"
)

type Actor struct {
	ID                 string      `json:"id"`
	WorkspaceID        string      `json:"workspace_id"`
	AuthorID           string      `json:"author_id"`
	Name               string      `json:"name"`
	Slug               string      `json:"slug"`
	Description        string      `json:"description"`
	Readme             string      `json:"readme"`
	Categories         []string    `json:"categories"`
	Tags               []string    `json:"tags"`
	IsPublic           bool        `json:"is_public"`
	Status             ActorStatus `json:"status"`
	IconURL            string      `json:"icon_url"`
	TotalRuns          int         `json:"total_runs"`
	TotalRevenueCents  int64       `json:"total_revenue_cents"`
	Rating             float64     `json:"rating"`
	RatingCount        int         `json:"rating_count"`
	CreatedAt          time.Time   `json:"created_at"`
	UpdatedAt          time.Time   `json:"updated_at"`
}

type ActorVersion struct {
	ID          string    `json:"id"`
	ActorID     string    `json:"actor_id"`
	Version     string    `json:"version"`
	DockerImage string    `json:"docker_image"`
	InputSchema interface{} `json:"input_schema"`
	ChangeLog   string    `json:"change_log"`
	IsLatest    bool      `json:"is_latest"`
	PublishedAt time.Time `json:"published_at"`
}

File: packages/types/src/actor.ts
export type ActorStatus = 'DRAFT' | 'PUBLISHED' | 'DEPRECATED'

export interface JSONSchemaProperty {
  type: 'string' | 'number' | 'boolean' | 'array' | 'object'
  title?: string
  description?: string
  default?: unknown
  enum?: unknown[]
  items?: JSONSchemaProperty
  properties?: Record<string, JSONSchemaProperty>
  minimum?: number
  maximum?: number
  minLength?: number
  maxLength?: number
}

export interface JSONSchema {
  $schema?: string
  type: 'object'
  title?: string
  description?: string
  properties: Record<string, JSONSchemaProperty>
  required?: string[]
}

export interface Actor {
  id: string
  name: string
  slug: string
  description: string
  readme: string
  categories: string[]
  tags: string[]
  isPublic: boolean
  status: ActorStatus
  iconUrl?: string
  totalRuns: number
  rating: number
  ratingCount: number
  author: {
    id: string
    name: string
    username: string
    avatarUrl?: string
  }
  latestVersion?: string
  createdAt: string
  updatedAt: string
}

export interface ActorVersion {
  id: string
  actorId: string
  version: string
  dockerImage: string
  inputSchema: JSONSchema
  changelog?: string
  isLatest: boolean
  publishedAt: string
}

export interface ActorInputSchema {
  title: string
  description?: string
  type: 'object'
  properties: Record<string, JSONSchemaProperty>
  required: string[]
}
